### ISSUE TITLE
Internal compiler errory when copying nested structs

### ISSUE THREAD

**Issue by Silveryard** [microsoft/DirectXShaderCompiler#4066]
The following code will not compile:
```
struct S1 {
	float a;
};

struct S2 {
	S1 s1;
	Texture2D<float> tex;
};

void main(float4 pos: SV_Position) {
	S1 s1;
	S2 s2;
	s2.s1 = s1;
}
```

Used arguments:
`dxc -E main -T ps_6_0 -HV 2021 Test.hlsl`

Compiler output:
`Internal compiler error: access violation. Attempted to read from address 0x0000000000000020`

After some experimenting, here are some changes to the code above that will result in no error:
- Not specifying -HV 2021
- Removing S2.tex. Other types like float or SamplerState are fine but Texture2D and Texture2D<T> are not
- Defining an assignment operator for S1 like this: `void operator=(S1 x) {
		a = x.a;
	}`

**Comment by jaebaek** [comment]
I will take a look. This seems to be related to the operator overloading.

**Comment by jaebaek** [comment]
This is fixed by #4063
